### PR TITLE
New foliage capitalization

### PR DESCRIPTION
**Pull request by TheShadowFerret**
#### Summary
None

#### Purpose of change

The proper nouns in the new foliage introduced by #54411 should be always be capitalized.

#### Describe the solution

Capitalize.

#### Describe alternatives you've considered

Bad grammar?

#### Testing

N/A

#### Additional context
https://en.wikipedia.org/wiki/Maianthemum_stellatum
https://en.wikipedia.org/wiki/Reynoutria_japonica
https://en.wikipedia.org/wiki/Jerusalem_artichoke

### CODE REVIEW

File: data/json/furniture_and_terrain/furniture-flora.json [hunk 1]
```diff
@@ -60,7 +60,7 @@
   {
     "type": "furniture",
     "id": "f_maianthemum_stellatum",
-    "name": "starry false solomon's seal",
+    "name": "starry false Solomon's seal",
```
Comment (NetSysFire) [inline data/json/furniture_and_terrain/furniture-flora.json:63]:
```suggestion
    "name": "Starry false Solomon's seal",
```

It appears that most if not all things are capitalized at the start.

File: data/json/items/comestibles/raw_veggy.json [hunk 2]
```diff
@@ -1108,7 +1108,7 @@
     "symbol": "%",
     "quench": 6,
     "calories": 25,
-    "description": "Some young japanese knotweed stems.  They taste a bit like rhubarb and contain good amounts of vitamin A.  Refreshing.",
+    "description": "Some young Japanese knotweed stems.  They taste a bit like rhubarb and contain good amounts of vitamin A.  Refreshing.",
```
Comment (NetSysFire) [inline data/json/items/comestibles/raw_veggy.json:1111]: Not 100% sure about this but I am no native speaker.

**Review by NetSysFire — APPROVED** [review]
I agree with most of this. As someone who values consistency, I am embarassed by this blunder, even if it is relatively minor.

File: data/json/furniture_and_terrain/furniture-flora.json (hunk 1 above)
Comment (TheShadowFerret) [reply data/json/furniture_and_terrain/furniture-flora.json:63]: Items aren't capitalized in their ``"name"`` unless they're a proper noun

File: data/json/items/comestibles/raw_veggy.json (hunk 2 above)
Comment (TheShadowFerret) [reply data/json/items/comestibles/raw_veggy.json:1111]: Japanese is a proper noun

**Pull request merged**